src/pkg/cluster/injector.go

```diff
@@ -473,10 +473,8 @@ func (c *Cluster) getImagesAndNodesForInjection(ctx context.Context) (imageNodeM
 					continue
 				}
 
-				for _, taint := range nodeDetails.Spec.Taints {
-					if taint.Effect == corev1.TaintEffectNoSchedule || taint.Effect == corev1.TaintEffectNoExecute {
-						continue
```

Comment:
This continue does not skip the node, which is why I implemented the check as a function instead.